jobs/resource_job.go

```diff
@@ -406,10 +425,20 @@ func jobSettingsSchema(s *map[string]*schema.Schema, prefix string) {
 	}
 }
 
+func gitSourceSchema(s *map[string]*schema.Schema, prefix string) {
+	if p, err := common.SchemaPath(*s, prefix, "url"); err == nil {
+		p.ValidateFunc = validation.IsURLWithHTTPS
+	}
+	(*s)["tag"].ConflictsWith = []string{"branch", "commit"}
+	(*s)["branch"].ConflictsWith = []string{"commit", "tag"}
+	(*s)["commit"].ConflictsWith = []string{"branch", "tag"}
+}
+
 var jobSchema = common.StructToSchema(JobSettings{},
 	func(s map[string]*schema.Schema) map[string]*schema.Schema {
 		jobSettingsSchema(&s, "")
 		jobSettingsSchema(&s["task"].Elem.(*schema.Resource).Schema, "task.0.")
+		gitSourceSchema(&s["git_source"].Elem.(*schema.Resource).Schema, "")
```

Comment:
i haven't looked at the api spec yet, but why not calling `gitSourceSchema` from `jobSettingsSchema`, so that multi-task jobs are supported as well?

Reply:
It's a property of the job and not of the task, you can't have a mixed job git/not-git I think